osu.Game/Screens/Edit/Timing/TimingScreen.cs

```diff
@@ -132,6 +133,61 @@ protected override void LoadComplete()
                 }, true);
             }
 
+            protected override void Update()
+            {
+                base.Update();
+
+                trackActivePoint();
+            }
+
+            /// <summary>
+            /// Given the user has selected a control point group, we want to track any group which is
+            /// active at the current point in time which matches the type the user has selected.
+            ///
+            /// So if the user is currently looking at a timing point and seeks into the future, a
+            /// future timing point would be automatically selected if it is now the new "current" point.
```

Comment:
Maybe this is just my ignorance due to not being a mapper, but I'm not sure this UX is... desired? I expected this to be a barebones simple implementation that just highlights the latest control point group regardless of its composition, but this is a lot more complex than that. Does it need to be?

Reply:
If your working with timing you want to track only timing changes. If you're working on effect point the same applies.

Reply:
I follow that part of it but I'm not super sure that it works very well. Leads to a lot of hidden rules as to how things work and generally feels more convoluted than the naive option.

I guess we can let it go to user testing and find out what the users say about it - but I'll need to re-read the complex logic again because I kind of gave up on trying to follow it on the first read, hoping that maybe it could go away entirely...

Reply:
Also as a follow up thought this may work better if we can add a way to filter out groups on the list based on which control point type you want to be working on (so something like four toggles, and you deselect all but timing when working on timing).